source/postprocess/interface.cc

```diff
@@ -229,6 +229,24 @@ namespace aspect
             postprocessor_names.push_back (std_cxx11::get<0>(*p));
         }
 
+      // We want the 'global statistics' postprocessor to be always active, and always as
+      // first postprocessor (to get the columns in the statistics object right). Reorder
+      // and if necessary insert this postprocessor into the list.
+      if (std::find (postprocessor_names.begin(),
+                     postprocessor_names.end(),
+                     "global statistics") != postprocessor_names.end())
+        {
+          std::vector<std::string>::iterator position =
+            std::find (postprocessor_names.begin(),
+                       postprocessor_names.end(),
+                       "global statistics");
+          const std::string first_element = postprocessor_names[0];
+          postprocessor_names[0] = "global statistics";
+          *position = first_element;
```

Comment:
This algorithm is a bit dangerous because it reorders the other postprocessors when the user may have wanted them in a particular order.

I would rewrite this whole block as follows:
```
  // see if the user specified "global statistics" somewhere; if so, remove it from the list
  std::vector<std::string>::iterator new_end 
    = std::remove (postprocessor_names.begin(),
￼                             postprocessor_names.end(),
￼                             "global statistics");
  if (new_end != postprocessor_names.end())
    postprocessor_names.erase (new_end, postprocessor_names.end());

  // in any case, put the global statistics postprocessor at the front:
  postprocessor_names.insert(postprocessor_names.begin(), "global statistics");
```